src/home_robot/home_robot/agent/ovmm_agent/ovmm_agent.py

```diff
@@ -142,34 +156,34 @@ def _hardcoded_place(self):
 
     def act(self, obs: Observations) -> Tuple[DiscreteNavigationAction, Dict[str, Any]]:
         """State machine"""
-        vis_inputs = self._get_vis_inputs(obs)
+        info = self._get_info(obs)
         turn_angle = self.config.ENVIRONMENT.turn_angle
 
         self.timesteps[0] += 1
         if self.states[0] == Skill.NAV_TO_OBJ:
             if self.skip_nav_to_obj:
-                self._switch_to_next_skill(e=0)
+                info = self._switch_to_next_skill(e=0, info=info)
```

Comment:
It looks like nothing is being returned in any of the conditions when a skill is to be skipped, in which case we default to returning the stop action at the end of the `act` method. If I understood this correctly, then this is the wrong behavior and we should return the action and info in each of the conditions.

Reply:
Yes, if  the current skill is to be skipped, we just move to the next skill in the same time step. STOP is only called if all skills skip.

Reply:
So the current behvaior (of not returning and trying the next if condition) is intended.